### PR TITLE
EES-2350 initial FE work for new create methodology (WIP)

### PR DESCRIPTION
**Pull request by amyb-hiveit**
- Removes the methodology section from the create / manage publication form
- Updates the methodology paths to include the publication 
- Adds Create methodology and Link external methodology buttons to publications on the admin dashboard
meth1
- Adds an inline form for adding / editing an external methodology
meth2
- Adds edit and remove external methodology buttons
meth3

### CODE REVIEW

File: src/explore-education-statistics-admin/src/pages/publication/components/PublicationForm.tsx [hunk 1]
```diff
@@ -26,12 +20,9 @@ interface FormValues {
   teamEmail: string;
   contactName: string;
   contactTelNo: string;
-  methodologyChoice: 'existing' | 'external' | 'none';
-  methodologyId?: string;
-  externalMethodology?: ExternalMethodology;
 }
 
-export type PublicationFormValues = OmitStrict<FormValues, 'methodologyChoice'>;
+export type PublicationFormValues = FormValues;
```
Comment (duncan-at-hiveit) [inline src/explore-education-statistics-admin/src/pages/publication/components/PublicationForm.tsx:25]: Looks like we don't need this type anymore, as it's just the same as FormValues

File: src/explore-education-statistics-admin/src/pages/publication/components/PublicationForm.tsx [hunk 2]
```diff
@@ -152,30 +68,13 @@ const PublicationForm = ({
...
-        case 'external':
-          await onSubmit({ ...values, externalMethodology });
-          break;
-        default:
-          await onSubmit(values);
-      }
-    },
-    errorMappings,
-  );
+  const handleSubmit = useFormSubmit(async ({ ...values }: FormValues) => {
```
Comment (duncan-at-hiveit) [inline src/explore-education-statistics-admin/src/pages/publication/components/PublicationForm.tsx:73]: Do you need the object spreading here?  Or could it just be:

`useFormSubmit(async (values: FormValues) =>`

File: src/explore-education-statistics-admin/src/routes/methodologyRoutes.ts [hunk 3]
```diff
@@ -11,28 +11,29 @@ export interface MethodologyRouteProps extends RouteProps {
 
 export type MethodologyRouteParams = {
   methodologyId: string;
+  publicationId: string;
 };
 
 export const methodologySummaryRoute: MethodologyRouteProps = {
-  path: '/methodologies/:methodologyId/summary',
+  path: '/publication/:publicationId/methodology/:methodologyId/summary',
```
Comment (duncan-at-hiveit) [inline src/explore-education-statistics-admin/src/routes/methodologyRoutes.ts:18]: This does tie us into always needing a Publication for a Methodology.  Maybe one to just check with Cam, to see if there's a need for "global" methodologies at all?

File: src/explore-education-statistics-admin/src/pages/admin-dashboard/components/MethodologyExternalLinkForm.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    const schema = Yup.object<FormValues>({
+      title: Yup.string().required('Enter an external methodology link title'),
+      url: Yup.string()
+        .required('Enter an external methodology URL')
+        .url('Enter a valid external methodology URL')
+        .test({
+          name: 'currentHostUrl',
+          message: 'External methodology URL cannot be for this website',
+          test: (value: string) =>
+            Boolean(value && !value.includes(window.location.host)),
```
[the remark this answers is not in the document]
Comment (amyb-hiveit) [reply src/explore-education-statistics-admin/src/pages/admin-dashboard/components/MethodologyExternalLinkForm.tsx:37]: It makes the ts linter sad if I remove it :(

**Pull request merged**